jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/sam/sync/SyncServerlessApplicationDialog.kt

```diff
@@ -0,0 +1,444 @@
+                        this.isSelected = createNewStack
+                        this.toolTipText = (message("serverless.application.sync.tooltip.createStack"))
+                    }.bindSelected(
+                        { createNewStack },
+                        {
+                            if (it) {
+                                createNewStack = true
+                                syncType = SyncType.CREATE
+                            }
+                        }
+                    )
+                        .actionListener { event, component ->
+                            if (syncType != SyncType.CREATE) {
+                                syncType = SyncType.CREATE
+                                refreshTemplateParametersAndTags()
+                            }
+                        }
+                    cell(stackNameField)
+                        .apply {
+                            this.horizontalAlign(HorizontalAlign.FILL)
+                            this.enabledIf(createStackButton.component.selected)
+                        }.bindText(::newStackName)
+                        .applyToComponent {
+                            this.toolTipText = message("serverless.application.sync.tooltip.createStack")
+                        }.validationOnApply { field ->
+                            if (!field.isEnabled) {
+                                null
+                            } else {
+                                validateStackName(field.text, stackSelector)?.let { field.validationInfo(it) }
+                            }
+                        }
+                }
+
+                row {
+                    val updateStackButton = radioButton(message("serverless.application.sync.label.stack.select"), false).applyToComponent {
+                        isSelected = !createNewStack
+                        this.toolTipText = (message("serverless.application.sync.tooltip.createStack"))
+                    }.bindSelected(
+                        { !createNewStack },
+                        {
+                            if (it) {
+                                createNewStack = false
+                                syncType = SyncType.UPDATE
+                            }
+                        }
+                    ).actionListener { event, component ->
+                        if (syncType != SyncType.UPDATE) {
+                            syncType = SyncType.UPDATE
+                            refreshTemplateParametersAndTags()
+                        }
+                    }
+
+                    cell(stackSelector).apply {
+                        this.horizontalAlign(HorizontalAlign.FILL)
+                        this.enabledIf(updateStackButton.component.selected)
+                        this.errorOnApply(message("serverless.application.sync.validation.stack.missing")) {
+                            it.isEnabled && (it.isLoading || it.selected() == null)
+                        }
```

Comment:
why not?

```suggestion
                    cell(stackSelector)
                        .horizontalAlign(HorizontalAlign.FILL)
                        .enabledIf(updateStackButton.component.selected)
                        .errorOnApply(message("serverless.application.sync.validation.stack.missing")) {
                            it.isEnabled && (it.isLoading || it.selected() == null)
                        }
```

Reply:
similar comment across the rest of the panel

Reply:
Removed unnecessary this.